### PR TITLE
[SYCL] Fix USM event clearing when calling queue::wait()

### PR DESCRIPTION
**Pull request by sergey-semenov**
Clear the USM events from the queue upon calls to wait() so that they
are not waited for multiple times.

Signed-off-by: Sergey Semenov <sergey.semenov@intel.com>

### CODE REVIEW

File: sycl/unittests/queue/wait.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  Mock.redefine<detail::PiApiKind::piEventGetInfo>(redefinedEventGetInfo);
+  Mock.redefine<detail::PiApiKind::piEventRetain>(redefinedEventRetain);
+  Mock.redefine<detail::PiApiKind::piEventRelease>(redefinedEventRelease);
+
+  context Ctx{Plt};
+  TestContext.reset(new TestCtx(Ctx));
+  queue Q{Ctx, default_selector()};
+
+  unsigned char *HostAlloc = (unsigned char *)malloc_host(1, Ctx);
+  Q.memset(HostAlloc, 42, 1);
```
[the remark this answers is not in the document]
Comment (sergey-semenov) [reply sycl/unittests/queue/wait.cpp:72]: No, we have a call to queue::wait on the line just below this one, before checking the value of `TestContext->NEventsWaitedFor`

File: sycl/unittests/queue/wait.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#include <CL/sycl.hpp>
```
Comment (alexbatashev) [inline sycl/unittests/queue/wait.cpp:1]: I believe other unit tests have some header in sources.

File: sycl/unittests/queue/wait.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+TEST(QueueWaitTest, USMEventClear) {
+  platform Plt{default_selector()};
+  if (Plt.is_host()) {
+    std::cout << "Not run on host - no PI events created in that case"
+              << std::endl;
+    return;
+  }
+
+  // TODO: Skip test for CUDA temporarily
+  if (detail::getSyclObjImpl(Plt)->getPlugin().getBackend() == backend::cuda) {
```
Comment (alexbatashev) [inline sycl/unittests/queue/wait.cpp:60]: Can we use backend generalization API here?

Comment (alexbatashev) [reply sycl/unittests/queue/wait.cpp:60]: Resolved offline

File: sycl/unittests/queue/wait.cpp (hunk 2 above)
Comment (sergey-semenov) [reply sycl/unittests/queue/wait.cpp:9]: Done

**Pull request merged**